src/app/clusters/service-area-server/service-area-delegate.cpp

```diff
@@ -0,0 +1,101 @@
+#include "service-area-delegate.h"
+#include "service-area-server.h"
+
+using namespace chip::app::Clusters::ServiceArea;
+
+bool Delegate::GetSupportedLocationById(uint32_t aLocationId, uint32_t & listIndex, LocationStructureWrapper & aSupportedLocation)
+{
+    bool ret_value = false;
+
+    listIndex = 0;
+
+    // simple linear iteration to find the location with the desired locationID.
+    while (GetSupportedLocationByIndex(listIndex, aSupportedLocation))
+    {
+        if (aSupportedLocation.locationID == aLocationId)
+        {
+            ret_value = true;
+            break;
+        }
+
+        ++listIndex;
+    }
+
+    return ret_value;
+};
+
+/**
+ * When we cannot guarantee that the restrictions imposed on the SupportedLocations, CurrentLocation and Progress attributes
```

Comment:
Doc comments should reside in the header. Change this to non-doc comment and place it inside the impl.

Reply:
I have revisited all of the code documentation in the Service Area cluster and remove unnecessary docs and improved others in [this commit](https://github.com/project-chip/connectedhomeip/pull/33991/commits/f1bb1166abfb7871ded52e577b484ace1c7a45be).